doc/overviews/helm-charts-release-process.md

```diff
@@ -0,0 +1,333 @@
+# Helm Charts Overview
+
+This document describes both the development workflow and production release process for Kuadrant Operator Helm charts, from local development to [Artifact Hub](https://artifacthub.io/packages/helm/kuadrant/kuadrant-operator) publication.
+
+## 1. Development Process
+
+### Overview
+
+During development, developers can build, test, and validate helm chart changes locally before submitting PRs. The development process focuses on iterating on chart templates, testing changes.
+
+### Development Workflow
+
+```mermaid
+sequenceDiagram
+    participant Dev as 👨‍💻 Developer  
+    participant Local as 💻 Local Environment
+    participant GitHub as 📦 GitHub PR
+    participant CI as 🔄 CI/CD
+
+    Note over Dev, CI: 🔧 Local Development
+    Dev->>Local: Edit chart files in charts/kuadrant-operator/
+    Dev->>Local: Run make helm-build
+    Local->>Local: • Set WASM shim image<br/>• Set operator image<br/>• Generate templates from kustomize<br/>• Update Chart.yaml versions
+    
+    Note over Dev, CI: ✅ Testing & Validation  
+    Dev->>Local: Run helm lint charts/kuadrant-operator
+    Dev->>Local: Run helm template for testing
+    Dev->>Local: Run make verify-helm-charts
+    Local->>Local: Validate chart structure and syntax
+    
+    Note over Dev, CI: 📤 Submit Changes
+    Dev->>GitHub: Create PR with chart changes
+    GitHub->>CI: Trigger automated validation
+    CI->>CI: Run make verify-helm-charts
+    CI->>CI: Lint charts and test rendering
+    CI->>GitHub: Report validation results
+```
+
+### Key Development Commands
+
+| Command | Purpose | Description |
+|---------|---------|-------------|
+| `make helm-build` | Build Charts | Generates templates from kustomize and updates Chart.yaml versions |
+| `make verify-helm-charts` | Validate | Runs linting, validation, and checks for chart updates |
+| `helm lint charts/kuadrant-operator` | Lint | Validates chart structure and syntax |
+| `helm template charts/kuadrant-operator` | Test Rendering | Tests template rendering without installation |
```

Comment:
I'll add these 2 as "other useful commands" in a different table below

Reply:
Ill remove these as they are also mentioned in the updated development workflow